### DISCUSSION TITLE
search within a bounding box instead of globally

### DISCUSSION THREAD

**Discussion by xinyuluo** [osm-search/Nominatim#2587] [Q&A] [answered]
Hi everyone. I have been using Nominatim for a while and would like to deploy it on my own server. I wonder if there's code that controls search scope? I think default is to search globally, but ideally when I deploy the tool, I'd like change it a bit so that I can search within a bounding box only when coordinate parameters are provided.

Any help is appreciated! 🙏

**Comment by mtmail** [comment, marked as answer]
You can use the `viewbox` and `bounded=1` search filters. https://nominatim.org/release-docs/latest/api/Search/#result-limitation

For example searching for all "Paris" in the eastern United States https://nominatim.openstreetmap.org/search.php?q=paris&viewbox=-79.81533%2C44.66899%2C-67.15908%2C36.73927&bounded=1&format=jsonv2

`bounded=0` on the other hand only affects the ranking. So results 1-5 are in the United States, result 6 is Paris in France.

You can also import only a country or region of the world https://nominatim.org/release-docs/latest/admin/Import/#choosing-the-data-to-import

**Reply by xinyuluo** [reply]
@mtmail thank you! do you know what part of source code is responsible for `viewbox` and `bounded` search filters?

**Reply by xinyuluo** [reply]
Also, I was reading the `bounded` section from the link you provided but didn't fully understand the following sentences.

> When `viewbox` and `bounded=1` are given, an **amenity only** search is allowed. Give the special keyword for the amenity in **square brackets**, e.g. [pub] and a selection of objects of this type is returned.

Perhaps could you help me clarify a few things here please?
1. It says amenity only search is allowed, but many results from the Paris example doesn't seem like amenities to me. So what have been excluded from the results of such queries?
2. If I understandably correctly, when `viewbox` and `bounded` are not present, there's no need to add a square bracket around query to do search for a specific category/tag (e.g. pub, restaurant, etc.). To clarify what it means to give the special keyword in square brackets, does it mean I need to write queries like `q=[pub]` instead of `q=pub`?
I tried to search for restaurants in singapore, and results for with/without square bracket look quite different but all points to some food places.
https://nominatim.openstreetmap.org/ui/search.html?q=restaurant&viewbox=103.6920359%2C1.1304753%2C104.0120359%2C1.4504753&bounded=1&limit=50
https://nominatim.openstreetmap.org/ui/search.html?q=[restaurant]&viewbox=103.6920359%2C1.1304753%2C104.0120359%2C1.4504753&bounded=1&limit=50

Thank you very much!

**Reply by mtmail** [reply]
`lib-php/SearchDescription.php` uses the parameters to create the database SQL query.

**Reply by mtmail** [reply]
> when viewbox and bounded are not present, there's no need to add a square bracket

I means then both square bracket or not the amenity-only search is ignored. Without the square brackets, the first URL you use, you get results where the title of a place contains 'restaurant', e.g. 'Chen Fu Ji Restaurant, 23, Smith Street, Chinatown'. It's just a coincidence it's all relevant places, many other restaurants (e.g. a McDonalds) won't be that in their title.

**Reply by xinyuluo** [reply]
@mtmail thanks for your reply. Sorry I didn't quite understand your last paragraph. As far as I see, there are also many places in the first URL that don't contain the term 'restaurant'?

**Reply by pawel-wroniszewski** [reply]
@xinyuluo For the record, the amenity-only search means searching for the amenity name given in the brackets, e.g. [pub]. It is allowed only when viewbox and bounded=1 are given. It does not mean, that you cannot do a regular search (for a name), when these params are given - it's just an option.
As for why you got some results of type 'amenity=restaurant' when searching for 'restaurant' - a different mechanism kicked in - 'special-phrases', described at the top here: https://nominatim.org/release-docs/latest/api/Search/#result-restriction and here: https://wiki.openstreetmap.org/wiki/Nominatim/Special_Phrases. The 'restaurant' is a special keyword which Nominatim translates to 'amenity=restaurant'. Both searches (for the name and for the amenity) are performed together, the results are later mixed together and ordered before returning to the user.
